### PR TITLE
added proposal for solidity2chaincode

### PR DESCRIPTION
**Pull request by salmanbaset**
Github link of current repo here:
https://github.com/AhmadZafarITU/sol2fab/

### CODE REVIEW

**Comment by tkuhrt** [comment]
@salmanbaset : If I remember correctly, this was a summer internship program, is that correct? Also, I wonder if the name of the lab should be more descriptive. Something like solidity_to_go_chaincode.

**Comment by salmanbaset** [comment]
@tkuhrt indeed, this was a project during summer internship program. I have updated lab name to `solidity_to_js_chaincode` .

**Comment by lehors** [comment]
I personally think the name is a bit too explicit now. :-) I would have been happy with something like Solidity2Fabric. But that's not a biggy and not worth arguing over. Names!!!....
+1 for me.

**Comment by tkuhrt** [comment]
@lehors : It is probably important that it be solidity_to_chaincode to represent the distinction between chaincode and solidity (especially given that Fabric now supports EVM/solidity through the new chaincode released with 1.3). I would be okay with dropping the language if we thought this lab might extend beyond just javascript to the other supported languages of chaincode in the future.

**Comment by salmanbaset** [comment]
@tkuhrt there is some discussion to use Typescript, but Typescript is really a form of javascript. @https://github.com/AhmadZafarITU/

**Comment by tkuhrt** [comment]
@vipinsun @trbs @ryjones : Any other comments or +1's for this lab?

**Comment by ryjones** [comment]
+1

File: labs/sol2fab.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+# Short Description
+This tool converts Solidity contract into Javascript chaincode through source-to-source translation for running them onto Hyperledger Fabric.
+
+# Scope of Lab
+The purpose of the lab is to perform a source-to-source translation of Solidity contracts into Javascript chaincode for running them into
+Hyperledger Fabric.
+
+# Initial Committers
+_Enter the Github IDs for the set of initial committers._
```
Comment (trbs) [inline labs/sol2fab.md]: Please remove this line :-)

File: labs/sol2fab.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+- https://github.com/ahmadzafar
+- http://github.com/salmanbaset
```
Comment (trbs) [inline labs/sol2fab.md]: Nit-picking I know, but lets use https :-)

**Comment by trbs** [comment]
Added few nitpicks on the PR, otherwise I agree with @lehors about the name and not wanting to argue over it.

+1

**Comment by tkuhrt** [comment]
@salmanbaset : Can you make the few minor changes that @trbs has requested? Then we can get this added as a lab.

**Comment by salmanbaset** [comment]
@tkuhrt updates incorporated. Please take a look. Thanks.

**Comment by tkuhrt** [comment]
https://github.com/hyperledger-labs/solidity2chaincode

**Comment by salmanbaset** [comment]
@tkuhrt thanks. Are we automatically added as committers on the lab? https://github.com/ahmadzafar needs to move the code in this location.

**Comment by tkuhrt** [comment]
The people specified in the initial committers list should have received an invite to be part of the team that can commit to this lab. If you already were part of another lab, then you would have been automatically added (e.g., Chris Ferris). So I believe both you (@salmanbaset) and Ahmad (@ahmadzafar) will have to accept the invite. Then you will have write access to the repository.

**Pull request merged**